### PR TITLE
#554 clean test dual responses design

### LINKED ISSUE #554: Clean up and test `DualResponsesDesign`
Do this after #553 
See https://github.com/Roche/crmPack/blob/main/R/Design-class.R#L424 and corresponding model `LogisticIndepBeta`

To do:
- [x] Clean up similarly as the other classes above in the file
    - [x] Add for simpler navigation in RStudio the lines with `# ClassName ----` and `## class ----` and `## constructor ----`
	- [x] includes changing to `snake_case` where it helps: here change `Effmodel` to `eff_model`
	- [x] documentation cleanup:
    	- [x] replace double `##` with `#`
    	- [x] remove `@keywords class` line
    	- [x] replace `\code{\linkS4class{ClassName}}` with ``[`ClassName`]``
    	- [x] Use ``@slot model (`type or class`)\cr`` for documenting the type or class of the slots
    	- [x] Use simple title `ClassName` followed by the lifecycle badge, followed by the description of the class
    	- [x] Add the `@aliases` line for the class definition
	- [x] Use the argument names for `setClass` and replace `representation()` with simple `c()`
	- [x] For `contains` just use simple string instead of list with a single string
	- [x] remove the `validObject` line
	- [x] user constructor:
		- [x] Move the example here, and for consistency rename the file to start with upper case letter, e.g. `Design` instead of `design`
		- [x] Use `@rdname ClassName-class` to put the constructor documentation in the same place as the class definition
		- [x] Use ``@param model (`type or class`)\cr see slot definition.`` for documenting the arguments
		- [x] Remove `@keywords methods` line
		- [x] Remove `@return` line for now 
		- [x] Use `@inheritDotParams RuleDesign`
- [x] Clean up the example code file
    - [x] Rerun and commit example file affected
- [x] Add unit tests to https://github.com/Roche/crmPack/blob/main/tests/testthat/test-Design-class.R
    - [x] compare other tests in that file and use same approach
- [x] Clean up the relevant method code file for updates needed linked to the clean-up of class
- [x] Do a search in other examples using the class and do changes needed linked to the clean-up of class
- [x] Do a search in vignettes using the class and make changes needed linked to the clean-up of class

### PR DESCRIPTION
**Pull request by dimitriskontosBAY**
Closes #554

### CODE REVIEW

File: R/Design-class.R [hunk 1]
```diff
@@ -409,61 +409,63 @@ DualResponsesSamplesDesign <- function(eff_model,
...
-##' @export
-##' @keywords class
+# DualResponsesDesign.R ----
+
+## class ----
+
+#' `DualResponsesDesign.R`
+#'
+#' @description `r lifecycle::badge("stable")`
+#' #'
```
Comment (danielinteractive) [inline R/Design-class.R:419]:
```suggestion
#'
```

File: R/Design-class.R [hunk 2]
```diff
@@ -409,61 +409,63 @@ DualResponsesSamplesDesign <- function(eff_model,
...
+#' This is a class of design based on DLE responses using the [`LogisticIndepBeta`] model
+#  and efficacy responses using [`ModelEff`]  model class
```
Comment (danielinteractive) [inline R/Design-class.R:421]:
```suggestion
#  and efficacy responses using the [`ModelEff`]  model class
```

File: R/Design-class.R [hunk 3]
```diff
@@ -409,61 +409,63 @@ DualResponsesSamplesDesign <- function(eff_model,
...
+#' without DLE and efficacy samples. It contains all slots in
+#' [`RuleDesign`] and [`TDsamplesDesign`] class objects.
```
Comment (danielinteractive) [inline R/Design-class.R:423]:
```suggestion
#' without DLE and efficacy samples. It contains all slots from the
#' [`RuleDesign`] and [`TDsamplesDesign`] classes.
```

**Review by danielinteractive — APPROVED** [review]
Thanks a lot @dimitriskontosBAY ! Almost perfect!! Just very few suggestions inside, afterwards good to merge

**Comment by danielinteractive** [comment]
Thanks @dimitriskontosBAY I think you might need to update the `simulate` method for `DualResponsesDesign`, see e.g. https://github.com/Roche/crmPack/blob/86441af0a5a1496529706cdf8bc4643c09b8d9c8/R/Design-methods.R#L2863 and replace `Effmodel` with the new slot name `eff_model`

**Comment by dimitriskontosBAY** [comment]
> Thanks @dimitriskontosBAY I think you might need to update the `simulate` method for `DualResponsesDesign`, see e.g.
> 
> https://github.com/Roche/crmPack/blob/86441af0a5a1496529706cdf8bc4643c09b8d9c8/R/Design-methods.R#L2863
> 
> and replace `Effmodel` with the new slot name `eff_model`

Thanks for the hint @danielinteractive! This is updated.

**Pull request merged**